### DISCUSSION TITLE
RollupError: Expected ident (Note that you need plugins to import files that are not JavaScript)

### DISCUSSION THREAD

**Discussion by kasunAththanagoda** [nuxt/nuxt#25002] [Questions]
i am trying to build my project using npm run build .but it gives me an error as 
RollupError: Expected ident (Note that you need plugins to import files that are not JavaScript)

i think the issue is related with ali oss package that i have installed and i can't figure out how to resolve this.
following is the full error i get   

ℹ Building Nitro Server (preset: node-server)                                                                                                      nitro 3:01:05 PM
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/client/getReqUrl.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/retry.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/setSTSToken.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/createRequest.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/encoder.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/isFunction.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/isBuffer.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/checkBucketName.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/obj2xml.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/omit.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/deepCopy.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/setRegion.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/checkConfigValid.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/object/getAsyncFetch.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/object/postAsyncFetch.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/deleteBucketInventory.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/putBucketInventory.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/getBucketWorm.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/getBucketInventory.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/extendBucketWorm.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/completeBucketWorm.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/listBucketInventory.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/getBucketStat.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/abortBucketWorm.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/bucket/initiateBucketWorm.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/policy2Str.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/checkEnv.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/formatObjKey.ts. Consider restricting the plugin to particular files via options.include
(inject plugin) rollup-plugin-inject: failed to parse C:/Users/Kasun Aththanagoda/Desktop/ican projects/git mall nuxt project/ican-mall-nuxt/node_modules/ali-oss/lib/common/utils/checkBucketTag.ts. Consider restricting the plugin to particular files via options.include

 ERROR  RollupError: Expected ident (Note that you need plugins to import files that are not JavaScript)                                            nitro 3:01:38 PM  

 6: import { checkConfigValid } from '../utils/checkConfigValid';
 7:
 8: export function getReqUrl(this: any, params) {
                              ^
 9:   const ep: any = {};
10:   const isCname = this.options.cname;

**Comment by TheAlexLichter** [comment]
Hey! Could you kindly provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by nevil2** [comment]
I have the same problem.  Appeared this morning after upgrading to 3.9.  Rollback to previous package-lock.json has not helped.  Now my application is broken.

The problem occurs importing my own module (which I've been doing for 2+ years).

```
[7:59:20 AM]  WARN  (inject plugin) rollup-plugin-inject: failed to parse ..../node_modules/rdb-utils/src/index.ts. Consider restricting the plugin to particular files via options.include

[nitro 7:59:20 AM]  ERROR  RollupError: Expected '{', got 'type' (Note that you need plugins to import files that are not JavaScript)

22: export * as grades from './utils/grades'
23: export type { GradeType, ClimbType } from './utils/grades'
           ^
24:
```
I am also getting warning on importing vue components from same library.
